src/Mono.Android/Android.Runtime/AndroidEnvironment.cs

```diff
@@ -344,18 +344,60 @@ static IWebProxy GetDefaultProxy ()
 #if !MONOANDROID1_0
 		[DynamicDependency (DynamicallyAccessedMemberTypes.PublicParameterlessConstructor, typeof (Xamarin.Android.Net.AndroidMessageHandler))]
 #endif
-		static object? GetHttpMessageHandler ()
+		static object GetHttpMessageHandler ()
 		{
-			var envvar = Environment.GetEnvironmentVariable ("XA_HTTP_CLIENT_HANDLER_TYPE")?.Trim ();
+			var handlerTypeName = Environment.GetEnvironmentVariable ("XA_HTTP_CLIENT_HANDLER_TYPE")?.Trim ();
 			Type? handlerType = null;
-			if (!String.IsNullOrEmpty (envvar))
-				handlerType = Type.GetType (envvar, false);
-			else
-				return null;
-			if (handlerType == null)
-				return null;
+			if (!String.IsNullOrEmpty (handlerTypeName))
+				handlerType = Type.GetType (handlerTypeName, throwOnError: false);
+
 			// We don't do any type checking or casting here to avoid dependency on System.Net.Http in Mono.Android.dll
-			return Activator.CreateInstance (handlerType);
+			if (handlerType is null || !IsAcceptableHttpMessageHandlerType (handlerType)) {
+				handlerType = GetFallbackHttpMessageHandlerType ();
+			}
+
+			return Activator.CreateInstance (handlerType)
```

Comment:
We should cache `handlerType` in a field, so that on repeated calls to `GetHttpMessageHandler()`, we don't need to redo all this lookup & validation.